content/docs/i18n.mdx

```diff
@@ -0,0 +1,243 @@
+
+:::
+
+
+The translate function has the following parameters:
+
+### `key`
+
+The key is used to identify the string, that needs to be translated.
+It can be set using the `UI Schema`, the `JSON Schema` or is generated by default based on the property path.
+
+#### UI Schema option
+
+The key can be set via the `i18n` UI Schema option:
+```json
+{ 
+  "type": "Control", 
+  "label": "name", 
+  "scope": "#/properties/name", 
+  "options": { 
+    "i18n": "customName" 
+  } 
+}
+```
+
+Therefore the translation will be invoked with `customName.label` & `customName.description`.
+
+#### JSON Schema
+
+The key can also be set with the custom JSON Schema `i18n` option:
+```json
+{
+  "name": { 
+    "type": "string", 
+    "i18n": "myCustomName" 
+  }
+} 
+```
+
+Therefore the translation will be invoked with `myCustomName.label` & `myCustomName.description`.
+
+#### Default: Property path
+
+Is none of the above is set, the property path will be used as the key.
+e.g. `name`.
+
+```
+{
+  properties: {
+    firstName: {
+      type: "string"
+    },
+    address: {
+      type: "object",
+      properties: {
+        street: {
+          type: "string"
+        }
+      }
+    }
+    comments: {
+      type: "array",
+      items: {
+        type: "object",
+        properties: {
+          message: {
+            type: "string"
+          },
+        }
+      }
+    }
+  }
+}
+```
+
+The paths in the above example would be:
+- `firstName.label` & `firstName.description`
+- `address.street.label` & `address.street.description`
+- `comments.message.label` & `comments.message.description` (the path for arrays will not contain indices)
+
+### `defaultMessage`
+
+The default message is provided by JSONForms and can act as a fallback or could be translated.
+
+:::note
+
+If the defaultMessage is `undefined` it should not be modified and still return as `undefined` (and not an empty string or similar).
+JSON Forms will use `undefined  when the message could be skipped or another generic key could be tried. 
+
+:::
+
+### `values`
+
+The values can contain additional context, if the key of the string is not enough to determine the correct translation.
+For example all error translations will have the error as part of the values object.
+
+## Error Customizations
+
+TODO: Description
+
+#### `<i18nkey>.error.custom`
+
+If `error.custom` is set, just a single error will be output, no matter how many errors are present.
+The `translate` function is called for `error.custom`
+The `translateError` function is called for each error.
+By default it behaves like this:
+
+## `translateError`
+
+The `translateError` is called for every AJV error. 
+`translateError` is an optional parameter to the i18n object. Usually it is not expected to customize this option.
+By default error it works like this:
```

Comment:
If no `<i18nkey>.error.custom` message is returned by the `translate` function, `translateError` will be called for each AJV error and the results combined.

The default implementation of `translateError` will invoke `translate` multiple times to determine the best message for the given error.
Therefore it's usually not necessary to customize `translateError` itself.